src/org/thoughtcrime/securesms/NicknameHandler.java

```diff
@@ -0,0 +1,70 @@
+package org.thoughtcrime.securesms;
+
+import android.content.Context;
+
+import org.thoughtcrime.securesms.database.DatabaseFactory;
+import org.thoughtcrime.securesms.database.RecipientDatabase;
+import org.thoughtcrime.securesms.recipients.Recipient;
+
+
+public class NicknameHandler {
+    private Context context;
+    private RecipientDatabase recipientDatabase;
+
+    public NicknameHandler(Context context) {
+        this.context = context;
+    }
+
+    public NicknameHandler setContext(Context context) {
+        this.context = context;
+        return this;
+    }
+
+    /**
+     * This method allows the developer to set the underlining database
+     * to any kind of database object passed in the parameter
+     * This method is implemented for only testing purposes inorder
+     * to pass mocked database object
+     * @param recipientDatabase
+     * @return
+     */
+    public NicknameHandler setRecipientDatabase(RecipientDatabase recipientDatabase) {
+        this.recipientDatabase = recipientDatabase;
+        return this;
+    }
+
+    /**
+     * This method responsible for setting the nick name
```

Comment:
Same comment about the first sentence.

Reply:
again, since it is javadoc, I was thinking to write something that is very clear and explains the reason for this methods existence

Reply:
it is clear without the need of the first sentence